### DISCUSSION TITLE
Docker result is not the same as using binary

### DISCUSSION THREAD

**Discussion by vitaly-zdanevich** [golangci/golangci-lint#3787] [Q&A]
From binary no errors, but from `docker run -t --rm -v $(pwd):/app -v ~/.cache/golangci-lint/v1.52.2:/root/.cache -w /app golangci/golangci-lint:v1.52.2 golangci-lint run -v` I got errors like this:

![image](https://user-images.githubusercontent.com/3514015/232599612-bac33fb6-27fb-4999-aae9-aea4ffc1283a.png)

**Comment by ldez** [comment]
Hello,

it's because you have a compilation problem.

`typecheck` is not a real linter it's just a way to parse/display "compilation" and linters errors (linter reports are not errors).

**Reply by vitaly-zdanevich** [reply]
Hi, but local binary of golangci-lint of the same version has no errors.

**Reply by ldez** [reply]
but inside a container, you don't have the same environment as locally:
- maybe you are using CGO and a system lib is missing
- maybe you have a problem with your dependencies
- maybe you don't mount the root of your project
- etc.

Based on your information, I can only say that the code analyzed by golangci-lint doesn't compile.

**Reply by vitaly-zdanevich** [reply]
I just ran that Docker command from the documentation, what else should I try?

> maybe you don't mount the root of your project

Hm, I did not do that..

**Comment by Antonboom** [comment]
@vitaly-zdanevich which Go are you using locally and how did you receive binary?

**Reply by Antonboom** [reply]
Probably related to https://github.com/golangci/golangci-lint/discussions/3400
